src/audio/tdfb/tdfb_generic.c

```diff
@@ -41,9 +43,18 @@ void tdfb_fir_s16(struct tdfb_comp_data *cd,
 		memset(cd->out, 0,  2 * out_nch * sizeof(int32_t));
 
 		/* Read two frames from all input channels */
+		c = 0;
 		for (i = 0; i < 2 * in_nch; i++) {
 			x = audio_stream_read_frag_s16(source, idx_in++);
 			cd->in[i] = *x << 16;
+			emp = iir_df2t(&cd->direction.emphasis[c++], *x << 16);
```

Comment:
Filtering and copy is missing from 32 bit function.

Reply:
Also the copy & filtering should happen only when tracking is enabled, not always.